ledgerwallet/manifest.py

```diff
@@ -99,74 +98,58 @@ def is_power2(n):
     return header + image_data
 
 
-class AppManifest(object):
-    def __init__(self, filename):
-        with open(filename) as f:
-            self.path = os.path.dirname(filename)
-            self.json = json.load(f)
-            assert "targetId" in self.json and "binary" in self.json
+class AppManifest(ABC):
+    dic: Dict = {}
 
-    @property
-    def app_name(self) -> str:
-        return self.json["name"]
+    def __init__(self, filename) -> None:
```

Comment:
As this `__init__` is empty and never called in children classes, it could be removed.